client/my-sites/stats/hooks/use-modeule-devices-query.ts

```diff
@@ -1,34 +1,64 @@
-import { useEffect } from 'react';
-import { useDispatch, useSelector } from 'calypso/state';
-import { getSiteSlug } from 'calypso/state/sites/selectors';
-import { requestMetrics } from 'calypso/state/stats/utm-metrics/actions';
-import { getMetrics, isLoading } from 'calypso/state/stats/utm-metrics/selectors';
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import getDefaultQueryParams from './default-query-params';
+
+export interface QueryStatsDevicesParams {
```

Comment:
Do you think it would make sense to extract all those TS types from all hooks to a single definition file?

Reply:
> extract all those TS types from all hooks

Do you mean all types defined under the `client/my-sites/stats/hooks` path?

Reply:
Yes, I wonder if this won't make importing in other files that use hooks easier.

Reply:
```
// types.d.ts
declare global {
	interface QueryStatsDevicesParams {
		date?: string;
		days?: number;
		max?: number;
	}
}

export interface QueryStatsDevicesParams {}
```

After declaring a global interface for Stat query params in the `types.d.ts` at the root of `client/my-sites/stats`, we wouldn't need to import it in every hook.

Reply:
MAybe that's something we could investigate for the more common types.